requirements.txt

```diff
@@ -10,7 +10,7 @@ pygtrie>=2.1,<3.0
 scipy>=1.4.1,<1.13.0
 sentencepiece>=0.1.91
 SoundFile; sys_platform == 'win32'
-torch>=2.1.0
-torchaudio>=2.1.0
+torch>=2.1.0,<=2.6.0
+torchaudio>=2.1.0,<=2.6.0
```

Comment:
I suppose the version cap is to prevent errors due to deprecated GradScaler, but I think I would prefer an approach that more smoothly transitions to the new GradScaler. Maybe we can warn that if you want to use GradScaler you have to use a more recent version of torch (I don't know when they introduced it).

Mainly, it is annoying for someone who wants to use the most recent version of torch to be forced to use an older version with SpeechBrain.

Reply:
Also, the new version allows someone to use a device other than cuda.

Reply:
> Also, the new version allows someone to use a device other than cuda.

I think you could still use gradscaler with a non-cuda device, cpu allows fp16 (which is a recent addition) but in practice people are using bf16 which doesn't requires gradscaler. Overall, I think I totally agree on what you suggested. What do you think of directly fetching the right gradscaler object based on the torch version? 

Something similar to checkpoint.py in which we would check the version of torch, and based on that, fetch the appropriate ,ethod.

Reply:
Sure, that makes sense to me!